File: comparison/results/haiku/ponytail/humaneval/HumanEval_1/python/solution.py
from typing import List


def separate_paren_groups(paren_string: str) -> List[str]:
    """ Input to this function is a string containing multiple groups of nested parentheses. Your goal is to
    separate those group into separate strings and return the list of those.
    Separate groups are balanced (each open brace is properly closed) and not nested within each other
    Ignore any spaces in the input string.
    >>> separate_paren_groups('( ) (( )) (( )( ))')
    ['()', '(())', '(()())']
    """
    cleaned = paren_string.replace(' ', '')
    
    groups = []
    current_group = ''
    open_count = 0
    
    for char in cleaned:
        current_group += char
        if char == '(':
            open_count += 1
        elif char == ')':
            open_count -= 1
        
        if open_count == 0 and current_group:
            groups.append(current_group)
            current_group = ''
    
    return groups

File: comparison/results/haiku/ponytail/humaneval/HumanEval_1/csharp/solution.cs
using System;
using System.Numerics;
using System.Diagnostics;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Security.Cryptography;
class Problem {
    // Input to this function is a string containing multiple groups of nested parentheses. Your goal is to
    // separate those group into separate strings and return the list of those.
    // Separate groups are balanced (each open brace is properly closed) and not nested within each other
    // Ignore any spaces in the input string.
    // >>> SeparateParenGroups(("( ) (( )) (( )( ))"))
    // (new List<string>(new string[]{(string)"()", (string)"(())", (string)"(()())"}))
    public static List<string> SeparateParenGroups(string paren_string) {
        List<string> result = new List<string>();
        string cleaned = paren_string.Replace(" ", "");
        
        int balance = 0;
        int start = 0;
        
        for (int i = 0; i < cleaned.Length; i++) {
            if (cleaned[i] == '(') {
                balance++;
            } else if (cleaned[i] == ')') {
                balance--;
            }
            
            if (balance == 0) {
                result.Add(cleaned.Substring(start, i - start + 1));
                start = i + 1;
            }
        }
        
        return result;
    }
}
